src/algo/k_shortest_path.rs

```diff
@@ -10,18 +10,28 @@ use crate::visit::{EdgeRef, IntoEdges, NodeCount, NodeIndexable, Visitable};
 /// \[Generic\] k'th shortest path algorithm.
 ///
 /// Compute the length of the k'th shortest path from `start` to every reachable
-/// node.
-///
-/// The graph should be `Visitable` and implement `IntoEdges`. The function
-/// `edge_cost` should return the cost for a particular edge, which is used
-/// to compute path costs. Edge costs must be non-negative.
+/// node. Edge costs must be non-negative.
 ///
 /// If `goal` is not `None`, then the algorithm terminates once the `goal` node's
 /// cost is calculated.
 ///
-/// Computes in **O(k * (|E| + |V|*log(|V|)))** time (average).
+/// # Arguments
+/// * `graph`: an input graph.
+/// * `start`: the *start* node.
+/// * `goal`: optional *goal* node.
+/// * `k`: sequence number of the required shortest paths.
+/// * `edge_cost`: closure that should return the cost for a particular edge, which is used
+///   to compute path costs. Edge costs must be non-negative.
+///
+/// # Returns
+/// * `HashMap`: [`struct@hashbrown::HashMap`] that maps `NodeId` to path cost.
+///
+/// # Complexity
+/// * Time complexity: **O(k(|E| + |V|log(|V|)))**.
+/// * Space complexity: **O(|V| + k|E|)**.
+///
+/// where **|V|** is the number of nodes and **|E|** is the number of edges.
```

Comment:
I think the complexity is slightly different:
```suggestion
/// # Complexity
/// * Time complexity: **O(k|E| log(k|E|))**.
/// * Space complexity: **O(|V| + k|E|)**.
///
/// where **|V|** is the number of nodes, **|E|** is the number of edges and **k** is the provided parameter.
```
For the space with the rationale that: In the worst case, put each of the |E| the maximum number of k times in the visit_next binary heap (first k |E| factor). Then each pop from the binary heap is log(#Entries) and the heap will then contain up to k |E| edges (second log ( k |E| ) factor).